src/openboardview/FileFormats/BRDFile.cpp

```diff
@@ -77,6 +77,27 @@ char *fix_to_utf8(char *s, char **arena, char *arena_end) {
 	return begin;
 }
 
+void BRDFileBase::AddNailsAsPins(const BRDPoint& max_point) {
+	for (auto &nail : nails) {
+		BRDPin pin;
+		pin.pos = nail.pos;
+		if (nail.side == BRDPartMountingSide::Both) {
+			pin.part = parts.size();
+			pin.side = BRDPinSide::Both;
+		} else if (nail.side == BRDPartMountingSide::Top) {
+			pin.part = parts.size();
+			pin.side = BRDPinSide::Top;
+		} else {
+			pin.pos.y = max_point.y - pin.pos.y;
```

Comment:
Also I think inverting y coordinate for bottom is specific to BRD2.

Reply:
Yes, thanks. This was unnoticed during gencad development and causing some problems there.
I moved the inverting from AddNailsAsPins to the process of BRD2 reading.
Tested on a BRD2 file with two-side nails that the result is unchanged.